src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -432,5 +438,73 @@ public IEnumerable<FilterLog> FindLogs(LogFilter filter, CancellationToken cance
 
             return error is null ? null : $"err: {error} (supplied gas {gasLimit})";
         }
+
+        public bool ExecuteWitness(BlockParameter blockParameter, Witness witness)
+        {
+            Block block = _blockTree.FindBlock(blockParameter);
+            RecoverTxSenders(block);
+            Block parent = _blockTree.FindBlock(block.ParentHash);
+            IBlockProcessor blockProcessor = _statelessBlocksProcessingEnv.GetProcessor(witness, parent.StateRoot);
+            Hash256 expectedHash = block.Hash;
+            Block[] result = blockProcessor.Process(parent.StateRoot, [block], ProcessingOptions.DoNotUpdateHead & ProcessingOptions.ReadOnlyChain, NullBlockTracer.Instance);
+            return expectedHash == result[0].Hash;
+        }
+
+        public Witness GenerateExecutionWitness(BlockParameter blockParameter)
+        {
+            Block block = _blockTree.FindBlock(blockParameter);
+            RecoverTxSenders(block);
+            Block parent = _blockTree.FindBlock(block.ParentHash);
+            (IBlockProcessor blockProcessor, WitnessGeneratingWorldState state) = _statelessBlocksProcessingEnv.CreateWitnessGeneratingBlockProcessor();
+            blockProcessor.Process(parent.StateRoot, [block], ProcessingOptions.DoNotUpdateHead & ProcessingOptions.ReadOnlyChain, NullBlockTracer.Instance);
+            HashSet<byte[]> stateNodes = new(new ByteArrayComparer());
+            HashSet<byte[]> codes = new(new ByteArrayComparer());
+            foreach ((Address account, HashSet<UInt256> slots) in state.StorageSlots)
+            {
+                AccountProofCollector accountProofCollector = new(account, slots.ToArray());
+                state.Accept(accountProofCollector, parent.StateRoot);
+                AccountProof accountProof = accountProofCollector.BuildResult();
+                codes.Add(state.GetCode(accountProof.CodeHash));
+                stateNodes.AddRange(accountProof.Proof);
+                stateNodes.AddRange(accountProof.StorageProofs.SelectMany(storageProof => storageProof.Proof));
+            }
+            Witness witness = new()
+            {
+                Keys = state.Keys,
+                Codes = codes.ToArray(),
+                Headers = [],
+                State = stateNodes.ToArray(),
+            };
```

Comment:
i think we can move this entirely in WitnessGeneratingWorldState
just a function - WitnessGeneratingWorldState.GenerateWitness - everything contined in WitnessGeneratingWorldState